tatsu/contexts.py

```diff
@@ -56,6 +56,11 @@ def wrapper(self):
     return decorator
 
 
+# This is used to mark left recursive rules
+def leftrec(impl):
+    impl.is_leftrec = True
```

Comment:
I realized that this should probably be part of `RuleInfo` instead. I don't know if `tatsumasu` can accept keyword arguments without interfering with its functionality, if it can then `@tatsumasu(leftrec=True)` would be the cleanest solution. Applying `@leftrec` first and then checking for `is_leftrec` in `tatsumasu` would work as well. I can make the necessary changes if needed.